src/Adapter/Contact/CommandHandler/AddContactHandler.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Contact\CommandHandler;
+
+use Contact;
+use PrestaShop\PrestaShop\Adapter\Domain\AbstractObjectModelHandler;
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\CleanHtml;
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\DefaultLanguage;
+use PrestaShop\PrestaShop\Core\Domain\Contact\Command\AddContactCommand;
+use PrestaShop\PrestaShop\Core\Domain\Contact\CommandHandler\AddContactHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Contact\Exception\CannotAddContactException;
+use PrestaShop\PrestaShop\Core\Domain\Contact\Exception\ContactConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Contact\Exception\ContactException;
+use PrestaShopException;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * Class AddContactHandler is used for adding contact data.
+ *
+ * @internal
+ */
+final class AddContactHandler extends AbstractObjectModelHandler implements AddContactHandlerInterface
+{
+    /**
+     * @var ValidatorInterface
+     */
+    private $validator;
+
+    /**
+     * @param ValidatorInterface $validator
+     */
+    public function __construct(ValidatorInterface $validator)
```

Comment:
well on the other hand its very reusable - don't see any reason to dislike it :thinking:

Reply:
@sarjon's point is that our Core, Domain and Adapter code should not rely on external dependencies such as the Symfony Validation component here.

It's a valid point, if you [google "do not depend external dependencies"](https://www.google.com/search?q=do+not+depend+external+dependencies) you'll find plenty of arguments on Sarjon's side 😄 .

I'm not sure I'd wish us to go there though 🤔 this would be another layer of complexity. It would mean create PrestaShop interfaces for every dependency we use, be it an Http Client, a Validator system, a FileSystem implementation, or even the CommandBus, then create adapters that would implement these interfaces and call external dependencies behind.

Something like
```
interface PrestashopHttpClientInterface
{
    public function doThis();
}

class GuzzleAdapter implements PrestashopHttpClientInterface
{
    public function __construct(GuzzleClient $client)
    {
        $this->client = $client;
    }

    public function doThis()
    {
        $this->client->doSomethingElse();
    }
}
```

That would be a very nice design, and if we need one day to replace on dependency for another it will be a lot easier because we would just need another implementation of the PrestashopHttpClientInterface. But this usecase happens not often so it's a trade-off between nice design and easy-to-use classes ...

I'd be happy to get some feedback from @mickaelandrieu @eternoendless @PierreRambaud @jolelievre here 😉

Reply:
>  or even the CommandBus

in fact, this is what we did with `CommandBus`, we defined custom interface and implemented adapter with Tactician.